merlin/models/tf/models/base.py

```diff
@@ -398,6 +402,11 @@ def _create_weighted_metrics(self, weighted_metrics=None):
                     for i, block in enumerate(self.model_outputs):
                         out[block.full_name] = weighted_metrics[i]
 
+        for metric in tf.nest.flatten(out):
+            # We need to ensure metrics passed to `compile()` are reset
+            if metric and metric.built:
```

Comment:
We need to check if metric is built to ensure related states variables were created (like in this [unit-test](class UpdateCountMetric(tf.keras.metrics.Metric):))

Reply:
Maybe we could remove this check for `metric.built` since in the keras code reset_metrics doesn't check this. And instead update the `UpdateCountMetric` `reset_state` to handle the absence of the `call_count` attribute.